Update docs (#430)
matrix-org/matrix-hookshot: docs/setup.md

Hookshot requires the homeserver to be configured with its appservice registration.

> ## Manual installation

Reviewer: For slightly more clarity (so folks looking for "yarn instructions" find it)

```suggestion
## Manual installation (via yarn)
```

[here](./setup/sample-configuration.md) for your convenience.

You should read and fill this in as the bridge will not start without a complete config.

> Sections and items marked as `Optional` need to either be configured if you want to use the respective feature, or removed/commented out to disable the feature.

Author: The wording "need" is not technically 100% precisely correct, but at least I think it works and is not wrong. See discussion at #431.

This remark could additionally be added to the respective service's pages, if required for visibility

Reviewer: "can" seems to fit here?

> Sections and items marked as `Optional` need to either be configured if you want to use the respective feature, or removed/commented out to disable the feature.

Reviewer: I want to be careful not to suggest the commenting out disables things, because that's not universally true.

```suggestion
Sections and items marked as `Optional` can either be configured if you want to use the respective feature, or removed/commented out. When items are commented out, the defaults are used.
```

- [Jira](./setup/jira.md)
- [Webhooks](./setup/webhooks.md)

> If you do not want to use a service, remove related configuration sections marked `Optional` to disable it.

Reviewer 2: ```suggestion
If you do not want to use a service, remove related configuration sections marked `Optional` to disable them.
```

        "retrying in 5s"
    ]
}
```

### Starting the bridge

> Finally after everything is configured, start the bridge using the method matching your installation method:

Reviewer 2: ```suggestion
Finally, after everything is configured, start the bridge using the method matching your installation method:
```

> Use `yarn start` to start a manually installed bridge and `docker run` or `docker-compose up -d` or similar with docker.

Reviewer 2: ```suggestion
Use `yarn start` to start a manually installed bridge and `docker run` or `docker-compose up -d` or similar with Docker.
```
